src/lpython/parser/semantics.h

```diff
@@ -19,19 +19,70 @@
 // everything from LFortran::AST to save typing:
 using namespace LFortran::LPython::AST;
 using LFortran::Location;
+using LFortran::Vec;
+
+static inline char* name2char(const ast_t *n) {
+    return down_cast2<Name_t>(n)->m_id;
+}
+
+static inline char** REDUCE_ARGS(Allocator &al, const Vec<ast_t*> args) {
+    char **a = al.allocate<char*>(args.size());
+    for (size_t i=0; i < args.size(); i++) {
+        a[i] = name2char(args.p[i]);
+    }
+    return a;
+}
+
+template <typename T, astType type>
+static inline T** vec_cast(const Vec<ast_t*> &x) {
+    T **s = (T**)x.p;
+    for (size_t i=0; i < x.size(); i++) {
+        LFORTRAN_ASSERT((s[i]->base.type == type))
+    }
+    return s;
+}
+
+#define VEC_CAST(x, type) vec_cast<type##_t, astType::type>(x)
+#define STMTS(x) VEC_CAST(x, stmt)
+#define EXPRS(x) VEC_CAST(x, expr)
 
 #define EXPR(x) (down_cast<expr_t>(x))
 #define EXPR2STMT(x) ((stmt_t*)make_Expr_t(p.m_a, x->base.loc, x))
 
 #define RESULT(x) p.result.push_back(p.m_a, EXPR2STMT(EXPR(x)))
```

Comment:
The error is fixed!!
```suggestion
#define STMT(x) (down_cast<stmt_t>(x))
#define RESULT(x) p.result.push_back(p.m_a, STMT(EXPR(x)))
```